libcaf_core/tests/legacy/detail/parser/read_number.cpp

```diff
@@ -56,7 +56,9 @@ struct range_consumer {
 
 struct res_t {
   std::variant<pec, double, int64_t> val;
-  template <class T>
+  res_t(const res_t&) = default;
+  res_t(res_t&&) = default;
+  template <class T, class E = std::enable_if_t<!std::is_same_v<T, res_t>>>
```

Comment:
```suggestion
  template <class T, class E = std::enable_if_t<!std::is_same_v<std::decay_t<T>, res_t>>>
```

Also, can we mark this `explicit` or would this break stuff?

Reply:
This one I remember trying with explicit and it breaks stuff. Since it's only used in this local file I might go and name the constructors.